xapian-applications/omega/docs/overview.rst

```diff
@@ -310,6 +310,11 @@ other filters too - see below):
 * MIME email messages (.eml) and USENET articles if perl with MIME::Tools and
   HTML::Parser is available
 * vCard files (.vcf, .vcard) if perl with Text::vCard is available
+* FictionBook v.2 files (.fb2, .fb2.zip) if libe-book is available
+* QiOO (mobile format, for java-enabled cellphones) files (.jar) if libe-book is available
+* TCR (simple compressed text format) files (.tcr) if libe-book is available
+* eReader files (.pdb) if libe-book is available
+* Sony eBook files (.lrf) if libe-book is available
```

Comment:
I'd expect EPUB files to be handled (https://en.wikipedia.org/wiki/EPUB) - they're one of the more popular ebook formats, and there appears to be a parser for them in the library: https://sources.debian.org/src/libe-book/0.1.3-1/src/lib/EPubParser.h/

Reply:
The list of types defined in the header is quite a bit longer than that in the README:

https://sources.debian.org/src/libe-book/0.1.3-1/inc/libe-book/EBOOKDocument.h/#L67

Some don't look useful (e.g. we handle HTML and XHTML already) but others do, assuming they are actually supported.

Reply:
Actually, a lot of that formats are not supported. Others (TYPE_EPUB, TYPE_HTML, TYPE_HTMLHELP, TYPE_OPENEBOOK, TYPE_ROCKETEBOOK, TYPE_XHTML) are only supported in an experimental way and it is probably that they appear in future versions.
[Here](https://sourceforge.net/projects/libebook/files/) is a list of formats.

Maybe I can add epub with libepub, but I am not sure if this is the best option.

Reply:
> Here is a list of formats.

That says "Source: README, updated 2013-11-16" which made me wonder if it was actually up to date, but looking at the git commits since then I don't see anything obviously either adding or firming up support for more formats.

The command-line tools from the library aren't packaged for Debian, but I tried loading several EPUB files into Libreoffice and it didn't handle them, so it does indeed seem EPUB isn't usefully supported.  That's rather a shame, since it seems to be the dominant format at this point.

> Maybe I can add epub with libepub, but I am not sure if this is the best option.

Another potential option is to make the EPUB support in libe-book work (at least well enough to get the text out without formatting, since that's what we care about here).  There is a parser class for it, so presumably the framework exists and it's a matter of implementing some missing methods within that:

https://sources.debian.org/src/libe-book/0.1.3-1/src/lib/EPubParser.cpp/

But let's get this merged before worrying about EPUB.